core/src/main/java/org/apache/iceberg/FileMetadata.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import java.nio.ByteBuffer;
+import java.util.Locale;
+import java.util.Map;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.iceberg.encryption.EncryptedOutputFile;
+import org.apache.iceberg.encryption.EncryptionKeyMetadata;
+import org.apache.iceberg.hadoop.HadoopInputFile;
+import org.apache.iceberg.io.InputFile;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.util.ByteBuffers;
+
+class FileMetadata {
```

Comment:
Ideally, we would name it `DeleteFiles` to be consistent with `DataFiles` but that's not possible. Do we want to move `DataFiles` into `FileMetadata` and deprecate it?

Reply:
Then we won't have to use methods from `DataFiles` here.

Reply:
I don't want to move anything in `DataFiles` in this commit because it's large enough already. I'm also thinking that this API may evolve further, so it's probably better to hold off on any compatibility breaking changes for now, until we know more about what this is going to look like.